src/components/shared/Entity/Actions/useSave.ts

```diff
@@ -201,6 +206,72 @@ function useSave(
         [backfillMode, collectionsBeingBackfilled.length, entityType]
     );
 
+    const updateResetOnCollections = useCallback(
+        async (
+            draftId: string,
+            collectionsOnDraft: DraftSpecsExtQuery_BySpecTypeReduced[] | null,
+            collectionsBeingRemovedFromDraft: any,
+            skipNonBackfilledCheck: boolean,
+            generateUpdatedSpec: (
+                draftSpec: DraftSpecsExtQuery_BySpecTypeReduced
+            ) => MassUpdateMatchData | null
+        ) => {
+            const collectionsToUpdate: MassUpdateMatchData[] = [];
+            if (collectionsOnDraft) {
+                collectionsOnDraft.forEach((draftSpec) => {
+                    // If we are removing it we do not need to update
+                    if (
+                        collectionsBeingRemovedFromDraft.includes(
+                            draftSpec.catalog_name
+                        )
+                    ) {
+                        return;
+                    }
+
+                    // skip = disable clean up
+                    // If the user enabled `data flow reset` and then totally turns off backfill
+                    //  when we are cleaning up we want to make sure we remove the `reset` prop
+                    //  so we do not care if the collection is not being backfilled or not
+                    if (!skipNonBackfilledCheck) {
```

Comment:
This is the secondary fix for issue `1` of https://github.com/estuary/ui/issues/1741